src/index.ts

```diff
@@ -511,120 +537,166 @@ export const plugin: Plugin<[FlexibleContainerOptions?], Root> = (options) => {
-      });
+          if (lastChild.type !== "text") return false;
 
-      if (!closingNode) return;
+          const fence = ":".repeat(fenceLen);
+          const re = new RegExp(`${fence}\\s*$`, "u");
+          return re.test(lastChild.value);
+        });
 
-      // just for type prediction
-      /* v8 ignore next */
-      if (!is<Paragraph>(closingNode, "paragraph")) return;
+        if (!closingNode) return;
 
-      const closingFlag = analyzeClosingNode(closingNode); // mutates the closingNode
+        // just for type prediction
+        /* v8 ignore next */
+        if (!is<Paragraph>(closingNode, "paragraph")) return;
 
-      const containerChildren = findAllBetween(
-        parent,
-        openingNode,
-        closingNode,
-      ) as BlockContent[];
+        const closingFlag = analyzeClosingNode(closingNode, fenceLen); // mutates the closingNode
 
-      if (openingFlag === "mutated") {
-        containerChildren.unshift(openingNode);
-      }
+        const containerChildren = findAllBetween(
+          parent,
+          openingNode,
+          closingNode,
+        ) as BlockContent[];
 
-      if (closingFlag === "mutated") {
-        containerChildren.push(closingNode);
-      }
+        if (openingFlag === "mutated") {
+          containerChildren.unshift(openingNode);
+        }
 
-      // if there is no content and type do not construct the container
+        if (closingFlag === "mutated") {
+          containerChildren.push(closingNode);
+        }
 
-      if (!containerChildren.length && !type) return;
+        // If there is no type, no title and no content between fences, treat as regular text
+        if (!type && !title && containerChildren.length === 0) {
+          return CONTINUE;
+        }
 
-      // if there is no content but type, then continue to construct the container
+        const titleNode = constructTitle(type, title, titleProps);
 
-      const titleNode = constructTitle(type, title, titleProps);
+        if (titleNode) containerChildren.splice(0, 0, titleNode);
 
-      if (titleNode) containerChildren.splice(0, 0, titleNode);
+        const containerNode = constructContainer(
+          containerChildren,
+          type,
+          title,
+          containerProps,
+        );
 
-      const containerNode = constructContainer(containerChildren, type, title, containerProps);
+        const { children } = parent;
+        const openingIndex = children.indexOf(openingNode);
+        const closingIndex = children.indexOf(closingNode);
+        children.splice(openingIndex, closingIndex - openingIndex + 1, containerNode);
+        changes++;
 
-      const { children } = parent;
-      const openingIndex = children.indexOf(openingNode);
-      const closingIndex = children.indexOf(closingNode);
-      children.splice(openingIndex, closingIndex - openingIndex + 1, containerNode);
+        return CONTINUE;
+      });
 
-      return CONTINUE;
-    });
+      if (changes === 0) break;
```

Comment:
The code logic so changed, I couldn't get it necessary `changes` if you remove ten times visit :)